### DISCUSSION TITLE
Systemair Topvex?

### DISCUSSION THREAD

**Discussion by cregeland** [GMTrevis/Homeassistant-NodeRed-Systemair-VTR300#15] [General]
Hi,

Would this work for a SystemAir Topvex TR20 unit? Of course the modbus registers needs to be modified. The Topvex unit also supports Modbus TCP.

Thanks :)

**Comment by GMTrevis** [comment]
@cregeland , sorry for my absence in this project due to lack of time.
Personally i would start all over when integrating the Topwex, there is most likely a lot of functionality that has to be adapted. 
The Topwex most likely have integrated outdoor compensation, you probably don't need my node red functionality for this, other then reading and writing to modbus registers (if you want that functionality integrated in HA). All temperature, moist and fan readings is most likely copy paste, if you change modbus addresses, IP and verify slave address.
Modbus registers for Topwex mode, alarms, override etc. most likely have to be adapted.

Find out what functionality you want to have in HA, then start out with the easiest modbus registers (input registers: temp., moist, fan speed etc). Copying and trying to use everything from my VTR300, will most likely cause an headache 😫, start small and systematic👍

Hope that helps you out.
Br 
GMT

**Reply by cregeland** [reply]
Thank you for your reply:)

I was hoping for just a simple integration with basic readout of fans speeds, temperature and filter status. On the control side only fan speed adjustment and temperature setpoint.

I guees I will need to find the modbus register and start with the basics😁

Thank you for your help😃

**Reply by cregeland** [reply]
Thanks for your input GMT,

I´ve got basic fan control functionality and external override (ventilator) working on the TopVex now - thanks to your work it was quite easy to modify the modbus codes and minor adjustments.

One question though - how did you integrate with the climate card? I would like to show actual supply temperature as well as set temperature in the the climate card. Looks like your flow is referencing an entity called sensor.climate.vtr300….. I only have climate.tr20 entity - no sensor.

I am able to set the temperature using an input_number entity, but I can´t seem to tie input_number to the climate card.

**Reply by GMTrevis** [reply]
Nice, my climate card approach was using a genereic thermostat (code below).
The "switch.vtr300_climate_dummyswitch" is not really being used, but i have configured it in node red to set a global variable "Heat assist on/off) in case i find it usefull some day. You need to create this dummy switch as the generic thermostat need a switch to set on/off to be properly configured.

When it comes to the setpoin adjustment from the climate card, i read the climate attribute (temp setpoint from the climate) in node red and compare it to the actual setpont temperature in the ventilation unit. If the setpoint attribute from the generic thermostat deviates from the setpoint in the ventilation unit, will the value from the generic thermostat be written to the ventilation unit supply setpoint temp. It should also work the opposite way, if changing the setpoint temp. on the ventilation unit, should the setpoint in the climate card update. I also have an option for outdoor temperature compensation (selected on/off), this is also included in the same "code", that's why it is a little more in my node red flow then you need.  

**Config for Generic thermostat:**
`- platform: generic_thermostat
  name: vtr300_tilluft_sp
  heater: switch.vtr300_climate_dummyswitch
  target_sensor: sensor.vtr300_tillufts_temperatur
  min_temp: 15
  max_temp: 30
  ac_mode: false
  cold_tolerance: 0.5
  hot_tolerance: 0.5
  min_cycle_duration:
    seconds: 60
  keep_alive:
    minutes: 30
  initial_hvac_mode: "off"
  away_temp: 22
  precision: 1 `

Screenshot of Node red config for the climate
![image](https://github.com/GMTrevis/Homeassistant-NodeRed-Systemair-VTR300/assets/69414078/fdb6f2d6-cd40-49d5-8c71-55d75036d7fa)

Screenshot of Node red for reading the generic themostat supply temp attribute
![image](https://github.com/GMTrevis/Homeassistant-NodeRed-Systemair-VTR300/assets/69414078/91e15291-2484-4b3d-9d77-bbd91f6122a9)

Climate attribute in template node
`{{ (states.climate.vtr300_tilluft_sp.attributes["temperature"] | float(0)) | round(1) }}`

I use the rotating heat exchanger 0-100% output to set/write the generic themostat climate state to visualize if the heat exchanger is on/off in the climate card. 

A little tricky way to implement the themostat to control the ventilation unit, but that was the best way i could do at the time. Still works, so i don't have any intention of changing this.

I think that's pretty much it, hope this helps you out.
